### PR TITLE
Show absolute address in branch instruction disassembly

### LINKED ISSUE #172: Show absolute address for branch instructions
Currently we show relative address which is unreadable.

Let's do this conditionally in `MCInstPrinter`, taking into account existing `PrintBranchImmAsAddress` flag (and let's set this flag in lldb)

### PR DESCRIPTION
**Pull request by mrk-its**
fixes #172

### CODE REVIEW

**Comment by mysterymath** [comment]
It looks like this is causing test failures in mos-relocs.s; would you take a look?

**Comment by mrk-its** [comment]
> It looks like this is causing test failures in mos-relocs.s; would you take a look?
yes, taking a look.

**Review by mysterymath — APPROVED** [review]
Nice, this makes llvm-objdump -d much easier to read. LGTM.

File: lld/test/ELF/mos-relocs.s [hunk 1]
```diff
@@ -50,8 +50,8 @@ relnext:
 # RELOCS-NEXT: 00000001 R_MOS_PCREL_8            .R_MOS_PCREL_8+0x4
 # RELOCS-NEXT: 00000003 R_MOS_PCREL_8            .R_MOS_PCREL_8
 # CHECK-LABEL: section .R_MOS_PCREL_8:
-# CHECK: bpl $2
-# CHECK: bpl $fc
+# CHECK: bpl $100c5
```
Comment (mrk-its) [inline lld/test/ELF/mos-relocs.s:53]: Why these sections are located above 0xffff here, @mysterymath?

Comment (mysterymath) [reply lld/test/ELF/mos-relocs.s:53]: The default ELF linker script built into the linker chooses 0x10000 as its start address.

Comment (mysterymath) [reply lld/test/ELF/mos-relocs.s:53]: It looks like we can actually change this if need be.

**Pull request merged**